### PR TITLE
Ag fix routes

### PR DESCRIPTION
**Pull request by runandrew**
Fixes to all routes from Code Review, and added Admin and Self-or-Admin filters/detection to restrict access to certain routes.

### CODE REVIEW

File: server/api.js [hunk 1]
```diff
@@ -1,25 +1,23 @@
...
   .use('/products', require('./routes/products'))
   .use('/reviews', require('./routes/reviews'))
-  
+
 // Send along any errors
 api.use((err, req, res, next) => {
-  res.status(500).send(err)
-})
+  res.status(err.status || 500).send(err.message || err)
+});
```
Comment (runandrew) [inline server/api.js:18]: Did this to enable custom error messages

File: server/api.js [hunk 2]
```diff
@@ -1,25 +1,23 @@
...
+const db = require('APP/db');
+const api = module.exports = require('express').Router();
 
 api
   .get('/heartbeat', (req, res) => res.send({ok: true,}))
   .use('/auth', require('./auth/auth'))
   .use('/users', require('./routes/users'))
-
-  .use('/addresses', require('./routes/addresses'))
   .use('/carts', require('./routes/carts'))
```
Comment (parky22) [inline server/api.js]: We should also delete /carts, /payments (carts and payments models were deleted)

File: server/routes/reviews.test.js [hunk 3]
```diff
@@ -66,14 +66,14 @@ describe('/api/reviews', () => {
 
     it('GET all reviews for a user...confirming we have 1 reviews', () =>
         request(app)
-            .get(`/api/reviews/user/1`)
+            .get(`/api/reviews/?userId=1`)
```
Comment (parky22) [inline server/routes/reviews.test.js:69]: should this be /api/reviews?userId=1

same for line 76

File: server/api.js (hunk 2 above)
Comment (runandrew) [reply server/api.js]: done

File: server/routes/reviews.test.js (hunk 3 above)
Comment (runandrew) [reply server/routes/reviews.test.js:69]: done, good find!

**Pull request merged**